development-styleguide.md

```diff
@@ -0,0 +1,107 @@
+## 👨‍💻 DEVELOPMENT STYLE GUIDE
+
+A practical reason to use a style guide is that they help you write code.
+Furthermore, they show commons mistakes or scenario and how to handle them
+
+### Typescript
+
+#### Object modification. Why you should avoid it.
+
+- Due to its dynamic nature, JavaScript makes it extremely easy to modify objects that you do not own
+- Such practice should generally be avoided as it can lead to maintenance problems and can produce hard to find bugs
+- If you don't own it, don't modify it
+
+side effects, example
+
+```typescript
+type Person = {
+  name: string;
+  surname: string;
+  age: number;
+};
+
+function printNameMutation(p: Person) {
+  console.log(`${p.name} ${p.surname}`);
+  // this object mutation could broken software logic
+  p.name = "Rob";
+}
+
+function printName(p: Person) {
+  console.log(`${p.name} ${p.surname}`);
+}
+
+const david: Person = { name: "David", surname: "Smith", age: 20 };
```

Comment:
this example looks like it has more to do with "use pure functions". I'll replace with an example that does not use functions. ie. 

```ts
let o = { x: 2 }
o.x = 1; // avoid
// use instead
const a = {...};
const b = { ...a, x: 1 }
```